runtime/compiler/net/MessageBuffer.cpp

```diff
@@ -108,5 +143,21 @@ MessageBuffer::computeRequiredCapacity(uint32_t requiredSize)
       extendedCapacity *= 2;
    return extendedCapacity;
    }
-};
 
+void
+MessageBuffer::tryFreePersistentAllocator()
+   {
+   if (J9::PersistentInfo::_remoteCompilationMode != JITServer::CLIENT)
+      return;
+
+   OMR::CriticalSection cs(getTotalBuffersMonitor());
+
+   if ((_totalBuffers != 0) || (_allocator == NULL))
+      return;
+
+   _allocator->~PersistentAllocator();
```

Comment:
I am thinking that because we allocated the `_allocator` with `new`, we should delete it with `delete`, otherwise we may leak the small amount of memory used by `class PersistentAllocator`. `delete` should call the destructor as well.

Reply:
I see that in other parts of the code we do something like:
```
      allocator->~PersistentAllocator();
      TR::Compiler->rawAllocator.deallocate(allocator);
      if (_persistentMemory)
         TR::Compiler->rawAllocator.deallocate(_persistentMemory);
      _persistentMemory = NULL;
```